### PR TITLE
Use `MvcTextFixture` as much as possible

### PR DESCRIPTION
**Pull request by dougbu**
- #3066 
  - reduce `TestServer` -> `PhysicalFileProvider` -> `FileSystemWatcher` count enough to run with Core CLR on Linux
- remove use of `HttpClient.DefaultRequestHeaders`; any client change affects other tests
- remove use of `RequestBuilder` class; creates a per-test `HttpClient` and requires the `TestServer`
- updated a few expectations because `CommonTestEncoder` does JavaScript a bit differently
  - "JavaScriptEncode[[...]]" -> "JavaScriptStringEncode[[...]]"
- side benefit: xUnit reports functional tests execute for only ~12.4s; was >30s before this change

nits:
- touched-up some leftover `_app` &c declarations to be more explicit and minimize `using`s
- moved statements into correct sections of methods in `RoutingTests`
- removed `TestLoggerFactory` and related classes from `TagHelperSampleTest`

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs [hunk 1]
```diff
@@ -1,106 +1,55 @@
...
+                    "/Home/Edit/0",
+                    "/Home/Edit/0?Name=Bobby&Blurb=howdy&DateOfBirth=1999-11-30&YearsEmployeed=2",
+                    "/Home/Index",
+                };
             }
         }
 
-        private class TestDisposable : IDisposable
+        [Theory]
+        [MemberData(nameof(PathData))]
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs]: Two overlapping changes aren't easy to read here:
- don't need `TestLoggerFactory` plus `NullLoggerFactory`
- `Home_Pages_ReturnSuccess()` should never have been a single test with a loop in it

File: test/Microsoft.AspNet.Mvc.FunctionalTests/WebApiCompatShimBasicTest.cs [hunk 2]
```diff
@@ -19,21 +19,25 @@
 
 namespace Microsoft.AspNet.Mvc.FunctionalTests
 {
-    public class WebApiCompatShimBasicTest
+    public class WebApiCompatShimBasicTest : IClassFixture<MvcTestFixture<WebApiCompatShimWebSite.Startup>>
     {
         private const string SiteName = nameof(WebApiCompatShimWebSite);
         private readonly Action<IApplicationBuilder> _app = new WebApiCompatShimWebSite.Startup().Configure;
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/WebApiCompatShimBasicTest.cs]: A bug I haven't got to the bottom of (specific to Core CLR test runs) prevents changing 3 test methods in this class.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/WebApiCompatShimBasicTest.cs [hunk 3]
```diff
@@ -61,24 +61,21 @@ public class WebApiCompatShimBasicTest
                 content);
         }
 
-#if !DNXCORE50
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/WebApiCompatShimBasicTest.cs]: Just happened to notice this test could be enabled with Core-specific expectations.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/project.json [hunk 4]
```diff
@@ -1,83 +1,83 @@
...
-    "exclude": [
-        "wwwroot",
-        "node_modules",
-        "bower_components"
-    ]
+    "Microsoft.AspNet.Mvc.TestConfiguration": "1.0.0",
+    "Microsoft.AspNet.TestHost": "1.0.0-*",
+    "Microsoft.AspNet.WebUtilities": "1.0.0-*",
+    "Microsoft.Framework.Configuration.Json": "1.0.0-*",
+    "Microsoft.Framework.Logging.Testing": "1.0.0-*",
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/project.json]: Added this line. The rest is noise.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/FilteredDefaultAssemblyProviderFixtureOfT.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                : base(libraryManager)
+            {
+            }
+
+            protected override IEnumerable<Library> GetCandidateLibraries()
+            {
+                var libraries = base.GetCandidateLibraries();
+
+                // Filter out other WebSite projects
+                return libraries.Where(library => !library.Name.Contains("WebSite") ||
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/FilteredDefaultAssemblyProviderFixtureOfT.cs]: I realize this is unrelated to your change - but is there a fixed set of names we could pass to this method? This seems kinda fragile.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs [hunk 6]
```diff
@@ -2,62 +2,105 @@
...
+                ?.CreateDelegate(typeof(Action<IApplicationBuilder>), startupInstance);
+            if (configureApplication == null)
+            {
+                var configureWithLogger = (Action<IApplicationBuilder, ILoggerFactory>)startupTypeInfo
+                    .DeclaredMethods
+                    .FirstOrDefault(m => m.Name == "Configure" && m.GetParameters().Length == 2)
+                    ?.CreateDelegate(typeof(Action<IApplicationBuilder, ILoggerFactory>), startupInstance);
+                Debug.Assert(configureWithLogger != null);
+
+                configureApplication = application => configureWithLogger(application, NullLoggerFactory.Instance);
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs]: If all we do is pass in a `NullLoggerFactory`, couldn't we simply change these websites to not use this `Configure` method?

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs [hunk 7]
```diff
@@ -1,106 +1,55 @@
...
-                return new TestLogger();
-            }
-
-            public void Dispose()
-            {
-            }
-        }
-
-        private class TestLogger : ILogger
+        public static TheoryData<string> PathData
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs]: :100:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/WebApiCompatShimBasicTest.cs (hunk 2 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/WebApiCompatShimBasicTest.cs]: :+1:

**Comment by pranavkm** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/FilteredDefaultAssemblyProviderFixtureOfT.cs (hunk 5 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/FilteredDefaultAssemblyProviderFixtureOfT.cs]: Suggest this is fine as-is. Slightly-filtered set includes every referenced framework library, the functional test library itself, and all sample site libraries. That is, it excludes _only_ the other non-sample web sites. And the idea appears to be being inclusive enough to make the `IControllerTypeProvider` implementation work a bit.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs (hunk 6 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs]: We're talking about a perfectly valid `Startup` class. And it (`TagHelperSample.Web`) is a sample and therefore more likely than most sites to be used outside the functional tests.

**Comment by dougbu** [comment]
d03a851

**Pull request closed without merging**